### DISCUSSION TITLE
Add Collection View From Lables In MAUI

### DISCUSSION THREAD

**Discussion by MahdiElahi** [dotnet/maui#25735] [General] [answered]
Hi ,
i want create a collection view like this image for my show recent search term in search content page
![Capture](https://github.com/user-attachments/assets/f147280a-3ca2-4d36-8c79-13ea19389e4c)
please help me 
i want frame lable width set based on the text length on lable like image

i use this code but have some problems :
 - the spacing between frames  items is not good 
 -  when scroll to right or left the border of frame is damage (see in images)
 
```
 <CollectionView ItemsSource="{Binding Tags}" VerticalOptions="Start">
            <CollectionView.ItemsLayout>
                <LinearItemsLayout Orientation="Horizontal"   ItemSpacing="10" />
            </CollectionView.ItemsLayout>
            <CollectionView.ItemTemplate>
                <DataTemplate>
                    <Frame Padding="5,0" 
                HorizontalOptions="Fill"
                VerticalOptions="Fill"
                BackgroundColor="#FAF9F6"
                CornerRadius="20"
                HasShadow="False"
                WidthRequest="{Binding Width}">

                    </Frame>
                </DataTemplate>
            </CollectionView.ItemTemplate>
        </CollectionView>
```
        
        
first 
        
![image](https://github.com/user-attachments/assets/1eb0b142-8fbd-43a2-aede-f09828e8facd)

after scroll to right or left
![bef](https://github.com/user-attachments/assets/09a58c2b-421e-4ebb-99d6-0c908d4fde44)

**Comment by stephenquan** [comment, marked as answer]
Have you considered swapping CollectionView for FlexLayout+BindableLayout?

**Comment by MahdiElahi** [comment]
Thanks @stephenquan
